routes/clients.rb

```diff
@@ -80,13 +80,25 @@ def self.registered(app)
                       contact_email: params[:contact_email] || client.contact_email,
                       contact_name: params[:contact_name] || client.contact_name,
                       active: params[:active] || client.active)
+
         redirect "/clients/#{params[:id]}"
       end
 
       app.delete '/clients/:id/?' do
         # delete a client
         client = Client[id: params[:id]]
-        client.delete unless client.nil?
+        unless client.name == 'default'
+          unless client.projects.empty?
```

Comment:
This code could be a model method, `client.remove`, which would check if it owns projects and use a project model method `project.remove` int he project model, which would reassign resources using a `project.reassign_resources` method, then call `project.delete`. Once this is complete, the `client.remove` method would call `client.delete`.